pkg/controller/job/job_controller.go

```diff
@@ -441,16 +443,21 @@ func (jm *Controller) getPodsForJob(j *batch.Job) ([]*v1.Pod, error) {
 // it did not expect to see any more of its pods created or deleted. This function is not meant to be invoked
 // concurrently with the same key.
 func (jm *Controller) syncJob(key string) (bool, error) {
+	status := "success"
 	startTime := time.Now()
 	defer func() {
+		syncDuration := time.Since(startTime).Seconds()
+		controller.ReconciliationDurations.WithLabelValues(jobControllerName, status).Observe(syncDuration)
 		klog.V(4).Infof("Finished syncing job %q (%v)", key, time.Since(startTime))
 	}()
 
 	ns, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
+		status = "error"
```

Comment:
This looks really intrusive. If we have to set error status on every return from every from sync in every controller, we'll inevitably forget to at some point and report success when we actually failed.